cmd/eksctl-anywhere/cmd/createcluster.go

```diff
@@ -146,7 +149,7 @@ func (cc *createClusterOptions) createCluster(cmd *cobra.Command, _ []string) er
 	}
 	defer close(ctx, deps)
 
-	clusterSpec, err = deps.CreateClusterDefaulter.Run(ctx, clusterSpec)
+	clusterSpec.Cluster, err = deps.CreateClusterDefaulter.Run(ctx, clusterSpec.Cluster)
```

Comment:
so we don't currently update anything in the defaulter that is not the cluster. since the machine health check defaulter is created for both create and upgrade, i changed the signature post this comment https://github.com/aws/eks-anywhere/pull/6176#discussion_r1271315060

Reply:
Right, but that comment was about not needing the spec to set defaults from the reconciler. Not about changing the signature of the CLI default runner.